src/accountsdb/db.zig

```diff
@@ -3221,8 +3312,11 @@ pub fn writeSnapshotTarWithFields(
         std.debug.assert(account_file.length == file_info.length);
 
         try snapgen.writeAccountFileHeader(archive_writer_counted, file_slot, file_info);
-        try archive_writer_counted.writeAll(account_file.memory);
-        try snapgen.writeAccountFilePadding(archive_writer_counted, account_file.memory.len);
+
+        var fifo = std.fifo.LinearFifo(u8, .{ .Static = std.mem.page_size }).init();
+        try fifo.pump(account_file.file.reader(), archive_writer_counted);
+
+        try snapgen.writeAccountFilePadding(archive_writer_counted, (try account_file.file.stat()).size); // feels silly, should probably store this
```

Comment:
```suggestion
        try snapgen.writeAccountFilePadding(archive_writer_counted, try account_file.file.getEndPos()); // feels silly, should probably store this
```
Also, yeah; the file's "actual length field" was removed in a PR at one point since it was redundant with the mmap'd buffer's length - may make sense to add back now in order to avoid needing to do this, especially considering account files are immutable.

Reply:
e424ea6479c52f62572e0ab5eff02f509c63307a